### PR TITLE
Fixed some typos and improved some comments

### PR DESCRIPTION
**Pull request by hw-qiaolei**
I don't like typos, although I always type them.

### CODE REVIEW

**Comment by bgrant0607** [comment]
Thanks, @hw-qiaolei. As long as you're doing this...

Most likely the same issues exist in pkg/api/v1/types.go.

Also, the type and field documentation is extracted, exported via swagger, and converted to user documentation. As discussed in [another PR](https://github.com/kubernetes/kubernetes/pull/15350#discussion_r41797383), please either convert to json field names and plain english when describing types, or omit those names entirely, as we do for, say, the container image: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L910

See also #14924.

**Comment by hw-qiaolei** [comment]
Hello @bgrant0607  thanks for reviewing this. See my comments below:

1. `pkg/api/v1/types.go` is much better since some inconsistencies are already fixed. Howerver I also fixed some I have found, please check.
2. For API fields' comments normalization, it seems that you do not reach a final conclusion on how we should do that yet, and I suggest do not do this work in this PR. I have noticed the problem when reading the code and agree with what @thockin said in #14924 but this PR only wants to fix some inconsistencies(only in one struct, not all definitions in this file) and typos at present.

- According to you: `either convert to json field names and plain english when describing types, or omit those names entirely`
- According to [go doc commentary](https://golang.org/doc/effective_go.html#commentary): `The first sentence should be a one-sentence summary that starts with the name being declared`

If you have a final conclusion I'd loved to share some work for it(I believe the authors who defined these types are most qualified for this since they know each field well).

**Comment by bgrant0607** [comment]
Go doc conventions don't apply for API field and type comments.

**Comment by bgrant0607** [comment]
If you want to defer fixing the field and type names, I'm fine with that. I was just suggesting additional fixes that would be useful.

However, for an example of the problem, see:
https://github.com/kubernetes/kubernetes/blob/master/docs/api-reference/v1/definitions.html#L6137

**Comment by bgrant0607** [comment]
LGTM

**Comment by bgrant0607** [comment]
@k8s-bot unit test this

**Comment by bgrant0607** [comment]
ok to test

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Comment by wojtek-t** [comment]
@hw-qiaolei 
 The failure seems related:

    02:23:23 Verifying ./hack/../hack/verify-generated-swagger-docs.sh
    02:23:54 FAILED

Can you please take a look and fix it?

File: pkg/api/v1/types.go [hunk 1]
```diff
@@ -236,8 +236,8 @@ type VolumeSource struct {
 	// Glusterfs represents a Glusterfs mount on the host that shares a pod's lifetime.
 	// More info: http://releases.k8s.io/HEAD/examples/glusterfs/README.md
 	Glusterfs *GlusterfsVolumeSource `json:"glusterfs,omitempty"`
-	// PersistentVolumeClaimVolumeSource represents a reference to a
-	// PersistentVolumeClaim in the same namespace.
+	// PersistentVolumeClaim represents a reference to a
+	// PersistentVolumeClaimVolumeSource in the same namespace.
```
Comment (thockin) [inline pkg/api/v1/types.go]: The second part of this change is wrong.

```
PersistentVolumeClaim volumes represent a reference to a PersistentVolumeClaim in the same namespace.

File: pkg/api/types.go [hunk 2]
```diff
@@ -189,12 +189,12 @@ type VolumeSource struct {
...
 	// NFS represents an NFS mount on the host that shares a pod's lifetime
 	NFS *NFSVolumeSource `json:"nfs,omitempty"`
-	// ISCSIVolumeSource represents an ISCSI Disk resource that is attached to a
+	// ISCSI represents an ISCSI Disk resource that is attached to a
 	// kubelet's host machine and then exposed to the pod.
 	ISCSI *ISCSIVolumeSource `json:"iscsi,omitempty"`
 	// Glusterfs represents a Glusterfs mount on the host that shares a pod's lifetime
 	Glusterfs *GlusterfsVolumeSource `json:"glusterfs,omitempty"`
-	// PersistentVolumeClaimVolumeSource represents a reference to a PersistentVolumeClaim in the same namespace
+	// PersistentVolumeClaim represents a reference to a PersistentVolumeClaimVolumeSource in the same namespace
```
Comment (thockin) [inline pkg/api/types.go]: The second part of this change is wrong.

```
PersistentVolumeClaim volumes represent a reference to a PersistentVolumeClaim in the same namespace.

File: pkg/api/v1/types.go [hunk 3]
```diff
@@ -236,8 +236,8 @@ type VolumeSource struct {
 	// Glusterfs represents a Glusterfs mount on the host that shares a pod's lifetime.
 	// More info: http://releases.k8s.io/HEAD/examples/glusterfs/README.md
 	Glusterfs *GlusterfsVolumeSource `json:"glusterfs,omitempty"`
-	// PersistentVolumeClaimVolumeSource represents a reference to a
-	// PersistentVolumeClaim in the same namespace.
+	// PersistentVolumeClaim represents a reference to a
```
Comment (thockin) [inline pkg/api/v1/types.go]: for all of the api changes - the field name should be jsonStyle not GoStyle (which is not godoc compatible, sadly)

**Comment by a-robinson** [comment]
Removing lgtm until comments are addressed to avoid auto-merge.

File: pkg/api/v1/types.go (hunk 1 above)
Comment (hw-qiaolei) [inline pkg/api/v1/types.go]: @thockin I am a not quite understand what's wrong, I changed it according to the definition:
```go
PersistentVolumeClaim *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
```

File: pkg/api/v1/types.go (hunk 3 above)
Comment (hw-qiaolei) [inline pkg/api/v1/types.go]: @thockin since it is not the purpose of this PR, I will close this PR and open another one to address this issue.

**Comment by hw-qiaolei** [comment]
Hello @bgrant0607 @thockin @wojtek-t @a-robinson ,

Thanks for spending your valuable time to review this PR. Since use `jsonStyle` not `GoStyle` is not the purpose of this PR, I will close it. 

1. If we just to address the `jsonStyle` not `GoStyle` issue discussed in [Use the json field name other than Go type names](https://github.com/kubernetes/kubernetes/pull/15350#discussion_r41797383). It seems like a physical labor and I'd like to take it. - I will open another PR to address it.
2. If we want to address all the problems raised by @thockin in [Need an API documentation audit](https://github.com/kubernetes/kubernetes/issues/14924), I think I am not the most qualified person to do that (I believe the authors who defined these types are most qualified for this since they know each field well). - I will leave this work to someone else.

What your opinions?

**Comment by bgrant0607** [comment]
@hw-qiaolei Please just fix typos and incorrect field names. Do not introduce field names where there were none, since that was deliberate.

File: pkg/api/v1/types.go (hunk 1 above)
Comment (thockin) [inline pkg/api/v1/types.go]: PersistentVolumeClaimVolumeSource is an internal type name.  The comment is telling the user what this field does - it refers to a PersistentVolumeClaim object.  User's don't care what we name our mid-level structures, but PersistentVolumeClaim is a first-class API object.

File: pkg/api/v1/types.go (hunk 3 above)
Comment (thockin) [inline pkg/api/v1/types.go]: can you open an issue to track, and reference this issue from it?

File: pkg/api/types.go (hunk 2 above)
Comment (thockin) [inline pkg/api/types.go]: not fixed yet

**Comment by thockin** [comment]
This is a good PR, it just needs a couple tweaks.

**Comment by thockin** [comment]
ping?

**Comment by thockin** [comment]
Ping?  This is a decent PR, just needs to be cleaned up for merge...if you will not followup, please close it.

**Comment by fejta** [comment]
This PR has had no meaningful activity for multiple months. If it is still valid please rebase, push a new commit and reopen the PR. Thanks!

**Pull request closed without merging**